lib/requests/consent-request.js

```diff
@@ -66,11 +66,13 @@ class ConsentRequest extends AuthRequest {
   static async get (req, res) {
     const request = ConsentRequest.fromParams(req, res)
 
-    const appOrigin = request.getAppOrigin()
+    const appUrl = request.getAppUrl()
+    const appOrigin = `${appUrl.protocol}//${appUrl.host}`
+    const serverUrl = url.parse(req.app.locals.ldp.serverUri)
     // Check if is already registered or is data browser
     if (request.isUserLoggedIn()) {
       if (
-        appOrigin === req.app.locals.ldp.serverUri ||
+        (appUrl && appUrl.host.includes(serverUrl.host) && appUrl.protocol === serverUrl.protocol) ||
```

Comment:
server: `https://safe.com/`
app: `https://unsafe.com/`

server: `https://safe.com/`
app: `https://safe.com.evil.com/`

Reply:
Ah right

Reply:
Tests are really important for these security cases.

Reply:
Yep, that was a lapse in judgment on my part. If we get the green light on deploying going forward with this version I'll crank out some more tests.

Though thankfully this wouldn't have been a security hole as this only controls if the consent screen pops up, and not if the app is allowed.

Reply:
The logic should only reside in one place though.